src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -155,12 +155,16 @@ public static void RemoveMemoryPressure(long bytesAllocated)
             _RemoveMemoryPressure((ulong)bytesAllocated);
         }
 
-
         // Returns the generation that obj is currently in.
         //
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern int GetGeneration(object obj);
+        public static int GetGeneration(object obj)
+        {
+            ArgumentNullException.ThrowIfNull(obj);
+            return GetGenerationWorker(obj);
+        }
 
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        private static extern int GetGenerationWorker(object obj);
```

Comment:
Nit: Our docs suggest to use `Internal` suffix for disambiguation: https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/corelib.md#naming-convention